### PR TITLE
cache clear

### PR DESCRIPTION
**Pull request by Akash-rumsan**
## 🔗 Related Issue

- https://github.com/rahataid/rahat-ui/issues/2234

- [x] Added issue link

## 📌 Description

**Cause**

All changes are correct. Here's a summary of what was done:

Root cause: All AA project Zustand stores persist data to localStorage. When switching between projects, the stores retained the previous project's data. Since components read from these stores synchronously, they showed stale data during the brief window between mount and when the new query resolved.

Fix applied to 9 service files:
stakeholders.service.ts — useStakeholders: clears stakeholders and stakeholdersMeta when UUID changes
groups.service.ts — useStakeholdersGroups, useBeneficiaryGroups, useBeneficiariesGroups: clear respective group stores when UUID changes
activities.service.ts — useActivitiesCategories, useActivities: clear categories/activities stores when UUID changes
phases.service.ts — usePhases: clears phases store when UUID changes
stats.service.ts — usePhasesStats: clears stats store when UUID changes
trigger-statements.service.ts — useAATriggerStatements: clears triggers store when UUID changes
grievances.service.ts — useGrievancesList: clears grievances store when projectUUID changes; also added projectUUID to query key (it was missing, causing cross-project cache collisions)
vendors.service.ts — useAAVendorsList: same fix as grievances
funManagement/service.ts — useFetchTokenStatsStellar: clears fund stats store when projectUUID changes; added projectUUID to query key

**Cache Issue Fixed Pages** 

- Stakeholders list /Stakeholders  Groups 
- Grievance Page
- Trigger Statement Main Page
- Beneficiary Page
- Vendor Page
- Activities Page
- Fund Management (Token Overview)

---

## ✅ Checklist

- [x] No `console.log` or debug code left
- [ ] Proper error handling implemented
- [x] Loading states handled (if needed)
- [x] API calls handled safely
- [x] No unnecessary code or comments
- [x] Self-reviewed code for logic, edge cases, and potential issues

---

## 🎯 Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] UI update
- [ ] Refactor
- [ ] Performance improvement

---

## 🧪 How to Test

1. Navigate to the pages mentioned  in **Cache Issue Fixed Pages** section

---

## ⚠️ Notes for Reviewer

- Check properly

### CODE REVIEW

File: libs/query/src/lib/aa/activities/activities.service.ts [hunk 1]
```diff
@@ -83,6 +88,12 @@ export const useActivities = (uuid: UUID, payload: any) => {
     setActivitiesMeta: state.setActivitiesMeta,
   }));
 
+  useEffect(() => {
+    setActivities([]);
+    setActivitiesMeta({});
+    qc.removeQueries({ queryKey: ['activities'], exact: false });
```
Comment (subash-rumsan) [inline libs/query/src/lib/aa/activities/activities.service.ts]: Not a good approach, calling the same remove queries before this seems like a very hacky method.
The best approach would be to handle the query with project UUID through the react query package itself. If it is not happening figure out why and the reason for it.

Comment (subash-rumsan) [reply libs/query/src/lib/aa/activities/activities.service.ts]: If you removeQueries every time uuid changes, you've thrown away instant back/forward navigation between projects — the exact thing React Query is supposed to give you for free. Go to project A → B → A again, and A refetches from scratch instead of showing cached data instantly (or instantly-then-revalidating with staleTime).

Comment (subash-rumsan) [reply libs/query/src/lib/aa/activities/activities.service.ts]: @Akash-rumsan

Comment (Akash-rumsan) [reply libs/query/src/lib/aa/activities/activities.service.ts]: @subash-rumsan  checking and figuring it out

**Review by hemanta-rumsan — APPROVED** [review]
